src/room-position.ts

```diff
@@ -75,7 +75,7 @@ interface RoomPosition {
      */
     findClosestByPath<T extends _HasRoomPosition | RoomPosition>(
         objects: T[],
-        opts?: FindPathOpts & { filter?: any | string; algorithm?: string },
+        opts?: FindPathOpts & { filter?: (object: T) => boolean | FilterObject | string; algorithm?: FindClosestByPathAlgorithm },
```

Comment:
There is an issue with `filter?: (object: T) => boolean | FilterObject | string;` It assumes that all of ` boolean | FilterObject | string` are return types from the `(object: T) =>` function, while only the `boolean` is expected to be.

This could be changed to 
`filter?: ((object: T) => boolean) | FilterObject | string;`
or
`filter?: FilterFunction<any> | FilterObject | string;`

Reply:
`FilterFunction` right now is tied into `FindConstant` and `FindTypes` which isn't appropriate because a `RoomPosition` could be passed.

I fixed the function definition, added tests, and re-pushed.